spec.html

```diff
@@ -42237,89 +42232,77 @@ <h1>%TypedArray%.prototype.set ( _source_ [ , _offset_ ] )</h1>
             1. Return ~unused~.
           </emu-alg>
         </emu-clause>
 
-        <emu-clause id="sec-settypedarrayfromarraylike" type="abstract operation" oldids="sec-%typedarray%.prototype.set-array-offset">
+        <emu-clause id="sec-settypedarrayfromtypedarray" type="abstract operation" oldids="sec-%typedarray%.prototype.set-typedarray-offset">
           <h1>
-            SetTypedArrayFromArrayLike (
+            SetTypedArrayFromTypedArray (
               _target_: a TypedArray,
               _targetOffset_: a non-negative integer or +&infin;,
-              _source_: an ECMAScript language value, but not a TypedArray,
+              _source_: a TypedArray,
             ): either a normal completion containing ~unused~ or a throw completion
           </h1>
           <dl class="header">
             <dt>description</dt>
             <dd>It sets multiple values in _target_, starting at index _targetOffset_, reading the values from _source_.</dd>
           </dl>
           <emu-alg>
-            1. Let _targetRecord_ be MakeTypedArrayWithBufferWitnessRecord(_target_, ~seq-cst~).
-            1. If IsTypedArrayOutOfBounds(_targetRecord_) is *true*, throw a *TypeError* exception.
-            1. Let _targetLength_ be TypedArrayLength(_targetRecord_).
-            1. Let _src_ be ? ToObject(_source_).
-            1. Let _srcLength_ be ? LengthOfArrayLike(_src_).
-            1. If _targetOffset_ = +∞, throw a *RangeError* exception.
-            1. If _srcLength_ + _targetOffset_ > _targetLength_, throw a *RangeError* exception.
-            1. Let _k_ be 0.
-            1. Repeat, while _k_ &lt; _srcLength_,
-              1. Let _Pk_ be ! ToString(𝔽(_k_)).
-              1. Let _value_ be ? Get(_src_, _Pk_).
-              1. Let _targetIndex_ be 𝔽(_targetOffset_ + _k_).
-              1. Perform ? TypedArraySetElement(_target_, _targetIndex_, _value_).
-              1. Set _k_ to _k_ + 1.
+            1. Let _targetRecord_ be ? ValidateTypedArray(_target_, ~seq-cst~).
+            1. Let _srcRecord_ be ? ValidateTypedArray(_source_, ~seq-cst~).
+            1. Let _srcLength_ be TypedArrayLength(_srcRecord_).
+            1. If _srcLength_ + _targetOffset_ > TypedArrayLength(_targetRecord_), throw a *RangeError* exception.
+            1. Perform ? ValidateTypedArrayCompatibility(_target_, _source_).
+            1. Let _targetElementSize_ be TypedArrayElementSize(_target_).
+            1. Let _srcElementSize_ be TypedArrayElementSize(_source_).
+            1. Let _targetByteIndex_ be _target_.[[ByteOffset]] + (_targetElementSize_ × _targetOffset_).
+            1. Let _limit_ be _targetByteIndex_ + (_targetElementSize_ × _srcLength_).
+            1. Let _targetType_ be TypedArrayElementType(_target_).
+            1. Let _srcType_ be TypedArrayElementType(_source_).
+            1. If _srcType_ is _targetType_, then
+              1. NOTE: The bit-level encoding of the source data must be preserved.
```

Comment:
I don't know why this note is appropriate in this position. Also, this reads like a normative requirement but it's in a note.

Reply:
I love the new wording.